specs/integrations/synthetics.md

```diff
@@ -0,0 +1,119 @@
+2. Includes trace.id and parent.id in all the Step Elasticsearch (ES) documents for the journey.
+
+```ts
+// run journey
+npx @elastic/synthetics --apm_tracing_urls "elastic.co/*" 
+
+// example.journey.ts
+journey("elastic e2e", ({ page }) => {
+  step("home page", async() => {
+    await page.goto("https://www.elastic.co")
+  })
+  step("blog page", async() => {
+    await page.goto("https://www.elastic.co/blog")
+  })
+})
+```
+
+Example of the tracing information added to the ES documents for two steps in the journey:
+
+```json
+// Step - homepage
+{"type":"step/end","journey":{"name":"elastic e2e"},"step":{"name":"home page","index":1,"status":"failed","duration":{"us":17382122}}, "trace.id": "xxx"}
+{"type":"journey/network_info","journey":{"name":"elastic e2e"},"step":{"name":"home page","index":1},"http":{"request":{"url":"http://www.elastic.co/","method":"GET"}},"trace.id": "t1", "transaction.id": "tr1"}
+
+
+// Step - blog page
+{"type":"step/end","journey":{"name":"elastic e2e"},"step":{"name":"blog page","index":2,"status":"failed","duration":{"us":17382122}}, "trace.id": "xxx"}
+{"type":"journey/network_info","journey":{"name":"elastic e2e"},"step":{"name":"blog page","index":2},"http":{"request":{"url":"http://www.elastic.co/blog","method":"GET"}},"trace.id": "t1", "transaction.id": "tr2"}
+```
+
+With this tracing information available in the ES documents for each step's network requests, the Synthetics UI can link back to the individual backend transactions on the APM.
+
+#### HTTP Checks
+
+For the below HTTP monitor
+
+```yml
+# heartbeat.yml
+heartbeat.monitors:
+- type: http
+  id: test-http
+  urls: ["https://www.example.com"]
+  apm:
+    enabled: true
+```
+
+Heartbeat would add the `traceparent` header to the monitored URL and add the
+other tracing related information to the ES documents.
+
+```json
+{"event":{"action":"monitor.run"},"monitor":{"id":"test-http","type":"http","status":"up","duration":{"ms":112}}, "trace.id": "t1", "transaction.id": "tr1"}
+```
+
+It's important to note that there is no dedicated waterfall information for the HTTP checks in the Synthetics UI. Consequently, the linking here will directly take you to the APM transaction if the backend is also traced by Elastic APM or OTEL (OpenTelemetry)-based agents.
+
+**NOTE: The correlation remain applicable even if downstream services are traced by OpenTelemetry (OTEL)-based agents. This ensures a consistent and seamless tracing experience regardless of the underlying tracing infrastructure..**
+
+### Identifying Synthetics trace
+
+Synthetics monitor executions creates `rootless traces` as these traces are not
+reported to the APM server. To overcome this limitation on the APM UI, we need
+to identify the synthetics traces and explicity link them to the Synthetics
+waterfall view. 
+
+- `http.headers.user-agent`:
+  - Contains `Elastic/Synthetics` for all outgoing requests from Synthetis based monitors.
+
+There is a limitation with this approach
```

Comment:
Another challenge is that native OTel agents don't capture all HTTP headers by default, IINM.

Reply:
I believe we cant do much here except the users would see a rootless trace unless Synthetics itself is also instrumented using APM? 

Would you like it to be added to this limitation list?

Reply:
I don't think it's a huge concern but we should still add it to the limitations.
The concern is not necessarily about the trace being rootless. I think it's more about that we aren't able to tell synthetics requests apart from normal requests. I think that's a limitation we can live with but it doesn't allow us to build features like "show only requests from synthetics".

Reply:
The requirement level of the [`user_agent.original`](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/http/#common-attributes) attribute (which is the `Value of the HTTP User-Agent header sent by the client.`) is `recommended`. I also think this is trivial to capture, so I expect most OTel implementation will have this out of the box.

Furthermore, what we could also do is this: for some languages, it's likely we'll end up with custom OTel distributions - with that we have control on how the OTel agent is configured. To be on a safe-side, we could add something like this to this spec:

> Custom Elastic OTel agent distributions MUST capture the  [`user_agent.original`](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/http/#common-attributes) attribute in order to enable APM-Synthetic correlation.

In most cases, this will have no effect, since I expect vanilla OTel agents already doing so. Nevertheless with this we can mitigate the issue of not being able to recognize synthetic calls from OTel.